components/core/src/package/archive.rs

```diff
@@ -417,6 +419,77 @@ pub trait FromArchive: Sized {
+    pub ldflags:           Option<String>,
+    pub cflags:            Option<String>,
+}
+
+impl PackageArchiveInfo {
+    pub fn new(path: impl Into<PathBuf>) -> Result<Self> {
+        let src = path.into();
+        let mut archive = PackageArchive::new(src.clone())?;
+        let header = artifact::get_artifact_header(&src)?;
+        let ident = archive.ident()?;
+        Ok(PackageArchiveInfo { format_version:    header.format_version,
+                                key_name:          header.key_name,
+                                hash_type:         header.hash_type,
+                                signature_raw:     header.signature_raw,
+                                origin:            ident.origin.clone(),
+                                name:              ident.name.clone(),
+                                version:           ident.version,
+                                release:           ident.release,
+                                checksum:          archive.checksum().ok(),
+                                target:            archive.target()
+                                                          .expect("pkg info archive target"),
+                                is_a_service:      archive.is_a_service(),
+                                deps:              archive.deps().unwrap_or_default(),
+                                build_deps:        archive.build_deps().unwrap_or_default(),
+                                tdeps:             archive.tdeps().unwrap_or_default(),
+                                build_tdeps:       archive.build_tdeps().unwrap_or_default(),
+                                exposes:           archive.exposes().unwrap_or_default(),
+                                pkg_services:      archive.pkg_services().unwrap_or_default(),
+                                resolved_services: archive.resolved_services().unwrap_or_default(),
+                                manifest:
+                                    archive.manifest()
+                                           .map_or_else(|_| None, |v| Some(v.to_string())),
+                                svc_user:
+                                    archive.svc_user()
+                                           .map_or_else(|_| None, |v| Some(v.to_string())),
+                                config:            archive.config()
+                                                          .map(std::string::ToString::to_string),
+                                ld_run_path:       archive.ld_run_path()
+                                                          .map(std::string::ToString::to_string),
+                                ldflags:           archive.ldflags()
+                                                          .map(std::string::ToString::to_string),
+                                cflags:            archive.cflags()
+                                                          .map(std::string::ToString::to_string), })
```

Comment:
`ToString` has already been brought into this namespace, so all these can be simplified to `.map(ToString::to_string)`